src/lib.rs

```diff
@@ -75,16 +76,18 @@ pub enum HostType {
 
 impl From<c_uint> for HostType {
     fn from(value: c_uint) -> Self {
-        match value {
-            0 => Self::Domain,
-            1 => Self::IPV4,
-            2 => Self::IPV6,
-            _ => Self::Domain,
+        if value > 2 {
+            return Self::Domain;
         }
+
+        // Safety: Prior to transmuting we checked if value is bigger than 2
+        // and returned the default in that case.
+        unsafe { std::mem::transmute(value) }
```

Comment:
It appears that it is extremely unsafe. Ref: https://x.com/tenellous/status/1819506534675894296

Is there any benchmark result you could share with us, that makes this change/risk worthy to land?